lib/quicly.c

```diff
@@ -175,6 +185,31 @@ struct st_quicly_application_space_t {
     int one_rtt_writable;
 };
 
+enum en_quicly_pending_new_cid_state_t {
+    /**
+     * this entry is free for use
+     */
+    QUICLY_PENDING_NEW_CID_STATE_VACANT,
+    /**
+     * this entry is to be sent at the next round of send operation
+     */
+    QUICLY_PENDING_NEW_CID_STATE_READY_TO_SEND,
+    /**
+     * this entry has been sent and is waiting for ACK (or timeout)
+     */
+    QUICLY_PENDING_NEW_CID_STATE_INFLIGHT,
+};
+
+/**
+ * records information for sending NEW_CONNECTION_ID frame
+ */
+struct st_quicly_pending_new_cid_t {
```

Comment:
Regarding the name of the structure and the necessary states, please refer to my comments below.